pages/apply/submit/declaration.tsx

```diff
@@ -27,72 +31,108 @@ const Declaration = (): JSX.Element => {
   const mainResident = useAppSelector((s) => s.application.mainApplicant);
   const application = useAppSelector((store) => store.application);
 
+  const [loading, setLoading] = useState<boolean>(false);
+  const [userError, setUserError] = useState<string | null>(null);
+
   const submitApplication = async () => {
     const [isEligible, reasons] = checkEligible(application);
+    setUserError(null);
+    setLoading(true);
     if (!isEligible) {
       const reasonStrings = reasons.map((reason) =>
         getDisqualificationReasonOption(reason)
       );
       const reason = reasonStrings.join(',');
+      // this is just a notification, so for now we don't need to handle errors
       dispatch(sendDisqualifyEmail({ application, reason }));
-      dispatch(disqualifyApplication(application.id!));
-      router.push('/apply/not-eligible');
+
+      try {
+        console.log('hello');
+        await dispatch(disqualifyApplication(application.id!)).unwrap();
```

Comment:
`.unwrap()` here probably doesn't do anything unless chained with `then` and `catch`. Should the `router.push` happen inside `then` and errors handled in `catch` should the dispatch result in rejected action?

Reply:
looks like unwrap() will cause an error to be thrown in this case, so to say it wouldn't do anything was not correct